### PR TITLE
Add targets to check, test, and build on CI

### PR DESCRIPTION
**Pull request by dio**
This PR proposes cleanup to the Makefile, hence we have several targets as follows:

```console
$ make help
scanner
help       Describe how to use each target
test       Run all unit tests
build      Build the binary
license    Add license to files
lint       Lint all Go sources
format     Format all Go sources
check      Verify contents of last commit
clean      Ensure a clean build
```

This also makes sure files have the right license. A `commit` workflow is also added.

This implicitly updates the Go version requirements to be at least Go1.17 (but it is only required for tooling, not building the binary nor required APIs).

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

**Comment by dio** [comment]
@incfly I can add a release workflow (also on how we produce "dist" tarballs) in the next PR if you're OK with this. Also, more tests: e2e and coverage.

🙇🏾

File: .github/workflows/commit.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+jobs:
+  check:
+    name: check
+    runs-on: ubuntu-20.04
+    timeout-minutes: 90  # instead of 360 by default.
+    strategy:
+      fail-fast: false # don't fail fast as sometimes failures are operating system specific
+    steps:
+      - name: Cancel when duplicated
```
Comment (incfly) [inline .github/workflows/commit.yaml:35]: Not quite follow this. will this cause issue if we want to manually trigger the GH action run?

**Review by incfly — APPROVED** [review]
nice! thank you so much!

File: .github/workflows/commit.yaml (hunk 1 above)
Comment (dio) [reply .github/workflows/commit.yaml:35]: This happens when you have multiple pushes in a PR. For example, you push a commit and then push another one (while the first commit is still in flight on the CI (the CI job is still doing the verification)). GHA, by default, will not cancel the first one.

**Comment by dio** [comment]
Thank you, @incfly!

**Pull request merged**